src/util/argstr.rs

```diff
@@ -27,58 +28,47 @@ impl<'a> ArgStr<'a> {
     }
 
     pub(crate) fn contains_byte(&self, byte: u8) -> bool {
-        assert!(byte.is_ascii());
+        debug_assert!(byte.is_ascii());
```

Comment:
These should be normal `assert`. Similarily below.

Reply:
I don't think these should eb normal `assert`.

1. `ArgStr` is a inner structure. It's the developers that ensure it's correctness.
2. `ArgStr` has never promised that it's inner bytes is valid UTF-8, which means seperating on arbitary u8 is legal. Actually we don't even need the assertion here.

Reply:
@pksunkara is correct. I would keep this and others as a normal `assert!`s, as searching for anything that isn't ASCII would return a useless result. Changing them to `debug_assert!` due to this being an internal structure is okay with me if the Clap developers would prefer that. However, searching for non-ASCII bytes would still be wrong, as the encoding is unknown.